src/Lean/Elab/Binders.lean

```diff
@@ -145,8 +145,7 @@ private def registerFailedToInferBinderTypeInfo (type : Expr) (ref : Syntax) : T
   registerCustomErrorIfMVar type ref "failed to infer binder type"
 
 private def addLocalVarInfoCore (lctx : LocalContext) (stx : Syntax) (fvar : Expr) : TermElabM Unit := do
-  if (← getInfoState).enabled then
-    pushInfoTree <| InfoTree.node (children := {}) <| Info.ofTermInfo { lctx := lctx, expr := fvar, stx, expectedType? := none }
+  withLCtx lctx {} <| addTermInfo stx fvar
```

Comment:
How is this change connected to PR?

Reply:
I had to set the `elaborator` field, so I took the opportunity to simplify. But maybe I simplified it too much (with unclear overhead), I can revert after we have decided on what to do with `getElaborator`.